pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTFieldDeclaration.java

```diff
@@ -152,4 +179,39 @@ public JavaFieldSignature getSignature() {
 
         return signature;
     }
+
+
+    /**
+     * Returns an iterator over the ids of the fields
+     * declared in this statement.
+     */
+    @Override
+    public Iterator<ASTVariableDeclaratorId> iterator() {
+        return iterateIds(this);
+    }
+
+
+    /* only for LocalVarDeclaration */ static Iterator<ASTVariableDeclaratorId> iterateIds(Node parent) {
```

Comment:
I think, this should not be in FieldDeclaration.
How about moving this static helper to VariableDeclarator? We would still need to have it static and use a parent node, to get the list of declarators.

Or maybe we should create a common base class for FieldDeclaration and LocalVariableDeclaration? Then these nodes could have a instance method "iterateIds()"... the base class could define+implement already `Iterable<ASTVariableDeclaratorId>`....

Reply:
We also should add the new rule to https://github.com/pmd/pmd/blob/master/pmd-core/src/main/resources/rulesets/releases/670.xml

Reply:
For now, I think moving this to VariableDeclarator is the way to go. It's hidden anyway

Adding a base class would be nice, but FieldDeclaration is a type node, contrary to LocalVariableDeclaration (this will change in 7.0.0, as FieldDeclaration shouldn't be a type node, and LocalVariableDeclaration shouldn't be an access node). A java 8 mixin interface would be more appropriate I think

Reply:
:+1: